### PR TITLE
org.wso2.carbon.identity.mgt code refactored

### PR DESCRIPTION
**Pull request by cdwijayarathna**
Fixed Findbugs critical and major issues - 360
licence headers added

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java [hunk 1]
```diff
@@ -112,54 +121,54 @@ public IdentityMgtConfig(RealmConfiguration configuration) {
...
+            if (enableUserAccountVerificationProperty != null) {
+                this.enableUserAccountVerification = Boolean.parseBoolean(enableUserAccountVerificationProperty.trim());
             }
 
-            String userAccountVerificationRole = properties.
+            String userAccountVerificationRoleProperty = properties.
                     getProperty(IdentityMgtConstants.PropertyConfig.ACCOUNT_VERIFICATION_ROLE);
-            if (userAccountVerificationRole != null && userAccountVerificationRole.trim().length() > 0) {
-                this.userAccountVerificationRole = userAccountVerificationRole;
+            if (userAccountVerificationRoleProperty != null && userAccountVerificationRoleProperty.trim().length() > 0) {
```
Comment (pulasthi7) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java]: Use StringUtils

**Comment by cdwijayarathna** [comment]
@hpmtissera  suggested changes commited.

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/ChallengeQuestionProcessor.java [hunk 2]
```diff
@@ -182,10 +185,12 @@ public void setChallengeQuestions(ChallengeQuestionDTO[] questionDTOs) throws Id
 
         } catch (Exception e) {
             String msg = "No associated challenge question found for the user";
```
Comment (mohanvive) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/ChallengeQuestionProcessor.java]: Two question
1) Any reason for logging an exception only in debug mode. ?
2) Unnecessarily creating a variable even If debug mode is disabled. Please fix.

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/ChallengeQuestionProcessor.java [hunk 3]
```diff
@@ -228,7 +233,9 @@ public UserChallengesDTO getUserChallengeQuestion(String userName, int tenantId,
...
```
Comment (mohanvive) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/ChallengeQuestionProcessor.java]: Any reason for catching generic exception ? and why logging the exception only in debug mode ?

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfigException.java [hunk 4]
```diff
@@ -1,3 +1,21 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (mohanvive) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfigException.java]: Why year has mentioned as 2014 ?

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/RecoveryProcessor.java [hunk 5]
```diff
@@ -124,7 +124,9 @@ public NotificationDataDTO recoverWithNotification(UserRecoveryDTO recoveryDTO)
             }
 
         } catch (UserStoreException e) {
-            log.warn("No Tenant domain for tenant id " + tenantId);
+            if(log.isDebugEnabled()){
```
Comment (mohanvive) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/RecoveryProcessor.java]: Any reason for logging exception only in debug mode ?

**Pull request merged**